### DISCUSSION TITLE
dzsave not respecting "layout" parameter after upgrade

### DISCUSSION THREAD

**Discussion by prohtex** [libvips/libvips#3285] [Q&A] [answered]
Hi there,

I'm a huge fan of VIPS and have been using it extensively for years as an Imagemagick replacement and so much more. Thank you for this blessed tool!

One of my scripts uses dzsave to generate image pyramids. I recently did a routine upgrade of packages and I guess VIPS was upgraded several point versions.

This command used to result in the creation of a zoomify format directory with ImageProperties.xml and TileGroup subdirectories. Now, unfortunately, whatever layout options I specify-zoomify, google, whatever-a .dz file is created instead.

`/opt/local/bin/vips icc_transform example.jpg example.dz[layout=zoomify] srgb`

I do this because I need to do an ICC profile conversion, and it is much more convenient this way, rather than create an intermediate image and run dzsave off that.

I can verify that this same command under vips-8.11.4 works as expected. I'm running macports vips-8.13.3. Any help is much appreciated!

**Comment by jcupitt** [comment, marked as answer]
Hi @prohtex,

This is part of the new save-to-stream feature.

When you use `.dz` as an output target, it now writes a zip file by default. This lets you send the output down a pipe (for example), so you can stuff like upload to S3 during pyramid creation. For example:

```
$ vips icc_transform k2.jpg .dz[layout=zoomify] cmyk | cat > banana.zip
```

Will make a zipfile containing the zoomify pyramid. Your example is also making a zip:

```
$ vips icc_transform example.jpg example.dz[layout=zoomify] srgb
$ unzip example.dz
Archive:  example.dz
 extracting: example/TileGroup0/3-0-0.jpg  
 extracting: example/TileGroup0/3-5-0.jpg  
...
 extracting: example/TileGroup0/0-0-0.jpg  
 extracting: example/ImageProperties.xml  
 extracting: example/vips-properties.xml  
```

You can force filesystem output like this:

```
$ vips icc_transform ../k2.jpg example.dz[layout=zoomify,container=fs] cmyk
$ ls
example  example.dz
$ ls example
ImageProperties.xml  TileGroup0  vips-properties.xml
```

You get an annoying empty `example.dz` file, which is unfortunate, but there is a zoomify directory too.

**Reply by prohtex** [reply]
Amazing. Thank you John for such a speedy reply and for your tireless work on VIPS. Cheers.
